errtracker/errtracker.go

```diff
@@ -0,0 +1,86 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package errtracker
+
+import (
+	"bytes"
+	"crypto/sha512"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"labix.org/v2/mgo/bson"
+
+	"github.com/snapcore/snapd/arch"
+	"github.com/snapcore/snapd/httputil"
+	"github.com/snapcore/snapd/release"
+)
+
+var (
+	crashDbUrlBase = "https://daisy.ubuntu.com/"
+	machineID      = "/var/lib/dbus/machine-id"
+
+	timeNow = time.Now
+)
+
+func Report(snap, channel, errMsg string) (string, error) {
+	machineID, err := ioutil.ReadFile(machineID)
+	if err != nil {
+		return "", err
+	}
+	machineID = bytes.TrimSpace(machineID)
+	identifier := fmt.Sprintf("%x", sha512.Sum512(machineID))
+
+	crashDbUrl := fmt.Sprintf("%s/%s", crashDbUrlBase, identifier)
+
+	report := map[string]string{
+		"ProblemType":        "Snap",
+		"Architecture":       arch.UbuntuArchitecture(),
+		"DistroRelease":      fmt.Sprintf("%s %s", release.ReleaseInfo.ID, release.ReleaseInfo.VersionID),
```

Comment:
The test crashes you submitted had a lowercase "ubuntu" in the DistroRelease string, it'd be nice if they had a capital 'U'.